yt_dlp/extractor/icareus.py

```diff
@@ -0,0 +1,226 @@
+                            videos\.minifiddlers\.org
+                        )
+                        /.+/player/.*(?:assetId|eventId)=(?P<id>\d+).*'''
+    _TESTS = [{
+        'url': 'https://www.helsinkikanava.fi/fi_FI/web/helsinkikanava/player/vod?assetId=68021894',
+        'md5': 'ca0b62ffc814a5411dfa6349cf5adb8a',
+        'info_dict': {
+            'id': '68021894',
+            'ext': 'mp4',
+            'title': 'Perheiden parhaaksi',
+            'description': 'md5:295785ea408e5ac00708766465cc1325',
+            'thumbnail': 'https://www.helsinkikanava.fi/image/image_gallery?img_id=68022501',
+            'upload_date': '20200924',
+            'timestamp': 1600938300,
+        },
+    }, {  # Recorded livestream
+        'url': 'https://www.helsinkikanava.fi/fi/web/helsinkikanava/player/event/view?eventId=76241489',
+        'md5': '014327e69dfa7b949fcc861f6d162d6d',
+        'info_dict': {
+            'id': '76258304',
+            'ext': 'mp4',
+            'title': 'Helsingin kaupungin ja HUSin tiedotustilaisuus koronaepidemiatilanteesta 24.11.2020',
+            'description': 'md5:3129d041c6fbbcdc7fe68d9a938fef1c',
+            'thumbnail': 'https://icareus-suite.secure2.footprint.net/image/image_gallery?img_id=76288630',
+            'upload_date': '20201124',
+            'timestamp': 1606206600,
+        },
+    }, {
+        'url': 'https://asahitv.fi/fi/web/asahi/player/vod?assetId=89415818',
+        'only_matching': True
+    }, {
+        'url': 'https://hyvinvointitv.fi/fi/web/hyvinvointitv/player/vod?assetId=89149730',
+        'only_matching': True
+    }, {
+        'url': 'https://inez.fi/fi/web/inez-media/player/vod?assetId=71328822',
+        'only_matching': True
+    }, {
+        'url': 'https://www.permanto.fi/web/alfatv/player/vod?assetId=165743121',
+        'only_matching': True
+    }, {
+        'url': 'https://suite.icareus.com/fi/web/westend-indians/player/vod?assetId=47567389',
+        'only_matching': True
+    }, {
+        'url': 'https://videos.minifiddlers.org/web/international-minifiddlers/player/vod?assetId=1982759',
+        'only_matching': True
+    }]
+    _API2_PATH = '/icareus-suite-api-portlet/publishing'
+
+    def _real_extract(self, url):
+        maybe_id = self._match_id(url)
+        page = self._download_webpage(url, maybe_id)
+        video_id = self._search_regex(
+            r"_icareus\['itemId'\]='(\d+)'", page, "video_id")
+        api_base = self._search_regex(
+            r'var publishingServiceURL = "(http.*?)";', page, "api_base")
+        organization_id = self._search_regex(
+            r"_icareus\['organizationId'\]='(\d+)'", page, "organization_id")
+        token = self._search_regex(
+            r"_icareus\['token'\]='([a-f0-9]+)'", page, "token")
+
+        token2 = self._search_regex(
+            r'''data\s*:\s*{action:"getAsset".*?token:'([a-f0-9]+)'}''', page,
+            "token2", default=None, fatal=False)
```

Comment:
* avoid use of `.*?`/`.+?`. You can use `[^"]+` instead
* Use `\s*` instead of space
* Use double quotes